src/absenceTypes/index.ts

```diff
@@ -12,10 +15,10 @@ export class AbsenceTypesEndpoint extends BaseApi {
     const response = this._get<AbsenceType[]>(`${this.getResourceName()}/read`, { params });
     return responseHandlers.list(response);
   }
-  public readRaw(requestParams?: RequestParams<AbsenceType> | Object): Promise<ApiResponseOnSuccess<AbsenceType[]>> {
-    const params = requestParams instanceof RequestParams ? requestParams.getParams() : requestParams;
+  public async readRaw(requestParams: RequestParams<AbsenceType>): Promise<ReadRawResponse<AbsenceType>> {
+    const params = requestParams.getParams();
     const response = this._get<AbsenceType[]>(`${this.getResourceName()}/read`, { params });
-    return responseHandlers.toApiResponse(response);
+    return createReadRawResponse<AbsenceType>(createResourceResponse(await createRawApiResponse(response)), requestParams);
```

Comment:
Forgot to add an actual comment: this looks a bit unholy to me

Reply:
Yes, it is. it will be refactored later when we have it for more endpoints.